src/ucp/core/ucp_request.c

```diff
@@ -328,6 +334,7 @@ static void ucp_request_mem_invalidate_completion(uct_completion_t *comp)
                                                   send.state.uct_comp);
     uct_worker_cb_id_t prog_id = UCS_CALLBACKQ_ID_NULL;
 
+    ucs_assertv(req->send.ep->refcount > 0, "ep %p", req->send.ep);
```

Comment:
otherwise it may not exist already, or even be set to NULL with this PR

Reply:
> otherwise it may not exist already, or even be set to NULL with this PR

no, it shouldn't be set here to `NULL`, we set to `NULL` only for close_req
the idea of the PR to make the EP is valid at this point